packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -407,6 +428,32 @@ export class TreeCheckout implements ITreeCheckoutFork {
 
 	private readonly views = new Set<TreeView<ImplicitFieldSchema>>();
 
+	public readonly transaction: ITransaction;
+	private readonly transactions = new TransactionStack();
+	/**
+	 * After pushing a starting revision to the transaction stack, this branch might be rebased
+	 * over commits which are children of that starting revision. When the transaction is committed,
+	 * those rebased-over commits should not be included in the transaction's squash commit, even though
+	 * they exist between the starting revision and the final commit within the transaction.
+	 *
+	 * Whenever `rebaseOnto` is called during a transaction, this map is augmented with an entry from the
+	 * original merge-base to the new merge-base.
+	 *
+	 * This state need only be retained for the lifetime of the transaction.
+	 *
+	 * TODO: This strategy might need to be revisited when adding better support for async transactions.
+	 * Since:
+	 *
+	 * 1. Transactionality is guaranteed primarily by squashing at commit time
+	 * 2. Branches may be rebased with an ongoing transaction
+	 *
+	 * a rebase operation might invalidate only a portion of a transaction's commits, thus defeating the
+	 * purpose of transactionality.
```

Comment:
This sentence seems like an argument in favor or against something but it's not clear what. Is it making a case against rebasing transaction steps?

Reply:
I don't know - I haven't really looked at this very closely, I just moved it.